### PR TITLE
Use green sleep to help retry the request

### PR DESCRIPTION
**Pull request by chenglch**
As sleep may block the main thread, this commit compare the
timestamp in the main loop to avoid of the blocking.

Below is just the UT result, as the change of framework do not affect the output.
``` 
[root@briggs01 ~]# rflash mid05tor12cn05 -a /mnt/xcat/iso/openbmc/910.1740.20171003b_1740A/witherspoon.pnor.squashfs.tar
mid05tor12cn05: Uploading /mnt/xcat/iso/openbmc/910.1740.20171003b_1740A/witherspoon.pnor.squashfs.tar ...
mid05tor12cn05: Upload successful. Attempting to activate firmware: IBM-witherspoon-ibm-OP9_v1.19_1.52
mid05tor12cn05: rflash started, please wait...
mid05tor12cn05: Activating firmware update. 10%
mid05tor12cn05: Activating firmware update. 10%
mid05tor12cn05: Activating firmware update. 10%
mid05tor12cn05: Firmware update successfully activated
```

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -543,11 +568,34 @@ sub process_request {
                 if ($rc != 0) {
                     $wait_node_num--;
                 } else {
-                    $status_info{ $node_info{$node}{cur_status} }->{process}->($node, undef);
+                    if ($status_info{ $node_info{$node}{cur_status} }->{process}) {
+                        $status_info{ $node_info{$node}{cur_status} }->{process}->($node, undef);
+                    } else {
+                        xCAT::SvrUtils::sendmsg([1,"Internal error, plase the check the process handler for current status "
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Spelling error here, but I think we should change to... 

`Internal error, check the process handler for current status`

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 2]
```diff
@@ -543,11 +568,34 @@ sub process_request {
...
                 delete $child_node_map{$cpid};
             }
         }
+        my @del;
+        while (my ($k, $v) = each %node_wait) {
+            if (time() >= $v) {
+                if ($node_info{$k}{method} || $status_info{ $node_info{$k}{cur_status} }{method}) {
+                    gen_send_request($k);
+                } else {
+                    xCAT::SvrUtils::sendmsg([1,"Internal error, plase the check the rest handler for current status "
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: Spelling error here, but I think we should change to... 

`Internal error, check the REST handler for current status`

**Comment by whowutwut** [comment]
Resolves https://github.com/xcat2/xcat-core/issues/4032

**Comment by chenglch** [comment]
Thanks @whowutwut , I have upload #4151 to fix this spelling error.

Normally, the code logic should not reach there if the state machine of our design and implementation are correct. But if not, this message could print more debug information, so the user could open a new issue with this output and help the developer to locate the error.

**Pull request merged**